test/integrationtests/session_test.go

```diff
@@ -0,0 +1,231 @@
+package integrationtests
+
+import (
+	"io"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/bsv-blockchain/go-bsv-middleware/pkg/temporary/sessionmanager"
+	"github.com/bsv-blockchain/go-bsv-middleware/pkg/temporary/wallet"
+	walletFixtures "github.com/bsv-blockchain/go-bsv-middleware/pkg/temporary/wallet/test"
+	"github.com/bsv-blockchain/go-bsv-middleware/test/assert"
+	"github.com/bsv-blockchain/go-bsv-middleware/test/mocks"
+	ec "github.com/bsv-blockchain/go-sdk/primitives/ec"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSessionCreationOnHandshake(t *testing.T) {
+	// given
+	sessionManager := mocks.NewMockableSessionManager()
+	serverWallet := mocks.NewMockableWallet()
+	server := mocks.CreateMockHTTPServer(serverWallet, sessionManager, mocks.WithLogger).
+		WithHandler("/", mocks.IndexHandler().WithAuthMiddleware())
+	defer server.Close()
+
+	clientWallet := mocks.CreateClientMockWallet()
+	initialRequest := mocks.PrepareInitialRequestBody(clientWallet)
+
+	serverWallet.OnCreateNonceOnce(walletFixtures.DefaultNonces[0], nil)
+	serverWallet.OnCreateSignatureOnce(prepareExampleSignature(t), nil)
+	serverWallet.OnGetPublicKeyOnce(prepareExampleIdentityKey(t), nil)
+
+	// when
+	response, err := server.SendNonGeneralRequest(t, initialRequest.AuthMessage())
+
+	// then
+	require.NoError(t, err)
+	assert.ResponseOK(t, response)
+
+	session := sessionManager.GetSession(initialRequest.IdentityKey)
```

Comment:
Isn't another action? - So `when` 🤔 
```suggestion
        // when
	session := sessionManager.GetSession(initialRequest.IdentityKey)
	
	// then
```

And the same suggestion/question for other occurrences.